mycpp/marksweep_heap.cc

```diff
@@ -0,0 +1,114 @@
+#include "mycpp/runtime.h"
+
+void MarkSweepHeap::Init(int max_heap_bytes) {
+  this->max_heap_bytes = max_heap_bytes;
+}
+
+void* MarkSweepHeap::Allocate(int byte_count) {
+  void* Result = calloc(byte_count, 1);
+  assert(Result);
+
+  this->AllAllocations.insert(Result);
+
+  this->current_heap_bytes += byte_count;
+  if (this->current_heap_bytes > this->max_heap_bytes) {
```

Comment:
What is the rationale behind the collection policy?

I think this might be OK to start with, but using a number of bytes doesn't seem obvious

Also I wouldn't call it "max_heap_bytes" as there is no maximum !  It's more like `collection_threshold_` or something

Reply:
> What is the rationale behind the collection policy?

I typed something completely arbitrary.  We need to profile before we make decisions here.

> I think this might be OK to start with, but using a number of bytes doesn't seem obvious

I'm not sure how using a number of bytes is non-obvious.  You set a target heap size when you initialize the collector, then it tries to stay under that.  What would be more intuitive?

> Also I wouldn't call it "max_heap_bytes" as there is no maximum ! It's more like `collection_threshold_` or something

Good point, changed.